python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,1089 @@
+
+        # Use the (flux under the dipole)*5 for an estimate.
+        # Lots of testing showed that having startingFlux be too high was better than too low.
+        startingFlux = np.nansum(np.abs(diArr) - np.nanmedian(np.abs(diArr))) * 5.
+        posFlux = negFlux = startingFlux
+
+        # TBD: set max. flux limit?
+        gmod.set_param_hint('flux', value=posFlux, min=0.1)
+
+        if separateNegParams:
+            # TBD: set max negative lobe flux limit?
+            gmod.set_param_hint('fluxNeg', value=np.abs(negFlux), min=0.1)
+
+        # Fixed parameters (dont fit for them if there are no pre-sub images or no gradient fit requested):
+        if (rel_weight > 0. and fitBgGradient):
+            if bgGradientOrder >= 0:
+                gmod.set_param_hint('b', value=0.)
+                if separateNegParams:
+                    gmod.set_param_hint('bNeg', value=0.)
+            if bgGradientOrder >= 1:
+                gmod.set_param_hint('x1', value=0.)
+                gmod.set_param_hint('y1', value=0.)
+                if separateNegParams:
+                    gmod.set_param_hint('x1Neg', value=0.)
+                    gmod.set_param_hint('y1Neg', value=0.)
+            if bgGradientOrder >= 2:
+                gmod.set_param_hint('xy', value=0.)
+                gmod.set_param_hint('x2', value=0.)
+                gmod.set_param_hint('y2', value=0.)
+                if separateNegParams:
+                    gmod.set_param_hint('xyNeg', value=0.)
+                    gmod.set_param_hint('x2Neg', value=0.)
+                    gmod.set_param_hint('y2Neg', value=0.)
+
+        y, x = np.mgrid[bbox.getBeginY():bbox.getEndY(), bbox.getBeginX():bbox.getEndX()]
+        in_x = np.array([x, y]).astype(np.float)
+
+        # I'm not sure about the variance planes in the diffim (or convolved pre-sub. images
+        # for that matter) so for now, let's just do an un-weighted least-squares fit
+        # (override weights computed above).
+        weights = 1.
+        if self.posImage is not None and rel_weight > 0.:
+            weights = np.array([np.ones_like(diArr), np.ones_like(diArr)*rel_weight,
+                                np.ones_like(diArr)*rel_weight])
+
+        # Note that although we can, we're not required to set initial values for params here,
+        # since we set their param_hint's above.
+        # add "method" param to not use 'leastsq' (==levenberg-marquardt), e.g. "method='nelder'"
+        result = gmod.fit(z, weights=weights, x=in_x,
+                          verbose=verbose,
+                          fit_kws={'ftol': tol, 'xtol': tol, 'gtol': tol, 'maxfev': 250},  # see scipy docs
+                          psf=self.diffim.getPsf(),  # hereon: kwargs that get passed to genDipoleModel()
+                          rel_weight=rel_weight,
+                          footprint=fp)
+
+        # Probably never wanted - also this takes a long time (longer than the fit!)
```

Comment:
I should modify the comment from "never wanted" to "never wanted in production". It is useful for debugging though.